examples/kanji_info.rs

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+* SPDX-License-Identifier: Apache-2.0 Refer to the file "LICENSE" for details.
+* example illustrating how to access kanji and reading information for a word
+*******************************************************************************/
+use jmdict::{KanjiElement, ReadingElement};
+
+fn main() {
+    let input = "一日";
+    let count = jmdict::entries()
+        .filter(|e| {
+            let mut result = false;
+            if e.kanji_elements().any(|k| k.text == input) {
+                result = true;
+                println!("--- entry #{} ---", e.number);
+                let kanji_elements: Vec<KanjiElement> = e.kanji_elements().map(|ke| ke).collect();
+                for kanji in &kanji_elements {
```

Comment:
The magic of jmdict's iterator-based interface is that you don't need to materialize into a Vec most of the time. You can just do:
```suggestion
                for kanji in e.kanji_elements() {
```
without any extra allocations. Same for `e.reading_elements()` below.